### PR TITLE
Feature/form preview

### PR DESCRIPTION
**Pull request by davidjamesstone**
[Update preview link in form editor](https://github.com/DEFRA/forms-designer/pull/173/commits/a680932802f64a25856878a3fe6716471ac0b22a)

and 

[Update overview page with form status and preview links](https://github.com/DEFRA/forms-designer/pull/173/commits/0264476c46052332767960de3aff0ac337871895)

@colinrotherham I introduced a `partials` directory under the main `views` folder to split out a couple of `includes` specifically for the overview page. Let me know if you had other plans for this. It didn't feel like they should live alongside the `common/nunjucks/components`.

### CODE REVIEW

File: designer/server/src/views/forms/partials/overview-draft.njk [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{% from "govuk/components/tag/macro.njk" import govukTag %}
```
Comment (colinrotherham) [inline designer/server/src/views/forms/partials/overview-draft.njk:1]: Partials are good, thanks for adding support for them

Not a blocker, but these two partials (draft, live) lend themselves to being components + tests?

E.g. Make it once but tell it which "state" to render in:

```njk
{% if form.draft %}
  {{ appFormInformation({
    metadata: form,
    state: 'draft'
  }) }}
{% endif %}
```

If you need conditional array items, you've got `.push()` available in Nunjucks:

```njk
{% set items = [
  {
    {
      key: {
        text: "Example 1"
      },
      value: {
        html: "Value 1"
      }
    }
  }
] %}

{% if params.metadata.xxx %}
  {% set items = (items.push([
    key: {
      text: "Example 2"
    },
    value: {
      html: "Value 2"
    }
  ]), items) %}
{% endif %}
```

Can always consolidate the two of them at a later date

**Review by colinrotherham — COMMENTED** [review]
Looks good @davidjamesstone

Thanks for fixing all the URLs. I've added some little comments

Can I leave it with you to get a decision on slug-first URL paths?

Would be a shame to do this PR twice

File: designer/server/src/views/forms/overview.njk [hunk 2]
```diff
@@ -6,72 +6,20 @@
 {% from "grid-layout/macro.njk" import appGridLayout %}
 {% from "heading/macro.njk" import appHeading %}
 {% from "form-card/macro.njk" import appFormCard %}
-{% from "form-status/macro.njk" import appFormStatus %}
+{% from "govuk/components/tag/macro.njk" import govukTag %}
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply designer/server/src/views/forms/overview.njk]: Well spotted, thanks :pray:

File: designer/client/src/components/Page/Page.tsx [hunk 3]
```diff
@@ -147,7 +147,10 @@ export const Page = (props: {
         </button>
         <a
           title={i18n('Preview page')}
-          href={new URL(`/${slug}${page.path}`, previewUrl).toString()}
+          href={new URL(
+            `/preview/draft/${slug}${page.path}`,
+            previewUrl
+          ).toString()}
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply designer/client/src/components/Page/Page.tsx]: Resolved with e375fcd800138e702184ee8239b1bcb08daaf9b4

File: designer/server/src/views/forms/partials/overview-live.njk [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+{% set formCreated %}
+  {{ (form.live.createdAt if form.live.createdAt else "2024-03-01") | formatDate }}
+  by {{ form.live.createdBy.displayName | default("Nathanael Booker", true) }}
+{% endset %}
+
+{% set formPreviewLink %}
+  {% set linkHref = previewUrl + "/preview/live/" + form.slug | urlencode %}
+  <a href="{{ linkHref }}" class="govuk-link govuk-link--no-visited-state" target="_blank" rel="noreferrer">
+    {{ linkHref }} (opens in a new tab)
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply designer/server/src/views/forms/partials/overview-live.njk:15]: Resolved with e375fcd800138e702184ee8239b1bcb08daaf9b4

File: designer/server/src/views/forms/partials/overview-live.njk [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+{% endset %}
+
+{% set formCreated %}
+  {{ (form.live.createdAt if form.live.createdAt else "2024-03-01") | formatDate }}
+  by {{ form.live.createdBy.displayName | default("Nathanael Booker", true) }}
+{% endset %}
+
+{% set formPreviewLink %}
+  {% set linkHref = previewUrl + "/preview/live/" + form.slug | urlencode %}
+  <a href="{{ linkHref }}" class="govuk-link govuk-link--no-visited-state" target="_blank" rel="noreferrer">
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply designer/server/src/views/forms/partials/overview-live.njk]: Resolved with c62582e1826057fae9343ed465e0f81c8a949fed

**Review by colinrotherham — APPROVED** [review]

File: designer/server/src/views/forms/partials/overview-draft.njk (hunk 1 above)
Comment (davidjamesstone) [reply designer/server/src/views/forms/partials/overview-draft.njk:1]: Yes I have used this `push` njk trick before. It's ugly but it works I guess.

As soon as we need to reuse these in a different context, I'll consider a macro :+1:

**Pull request merged**